### ISSUE TITLE
Fatal crash upon dying in the nether.

### ISSUE THREAD

**Issue by TheRobberPanda** [CleverNucleus/Data-Attributes#18]
Here are the logs. Please contact me if you need more information, I'll be happy to help in any way I can

[crash-2022-07-15_12.57.46-client.txt](https://github.com/CleverNucleus/Data-Attributes/files/9120276/crash-2022-07-15_12.57.46-client.txt)

**Comment by TheRobberPanda** [comment]
Negative, this was in my multiplayer server.

**Comment by CleverNucleus** [comment]
Ah okay, follow up question: did just your game client crash, or did the server crash too? I see that the log you have provided is for the client, but just to confirm there were no crash logs on the server? 
Thanks.

**Comment by TheRobberPanda** [comment]
No crash logs on the server, no. If you want to I can add you on discord and share screens, i dont mind.
TVMAN#4127

**Comment by CleverNucleus** [comment]
That's okay, I should be able to fix soon with the information you've provided. Thanks!

**Comment by TheRobberPanda** [comment]
Sorry. I am getting another bug. This time it isn't a crash so it generates no crash report and it doesn't seem to appear in the latest.log either, since I have looked. Character gets stuck after changing dimensions through portals or execute command. It happens both in singleplayer and in multiplayer, however in multiplayer you can fix it by rejoining. I haven't found any mod that could be the culprit and I was thinking since data-attributes already has a bug related to dimensions, then perhaps it could be the culprit. How can I help fix this issue?

**Comment by CleverNucleus** [comment]
Hi there,

> Character gets stuck after changing dimensions through portals or execute command.

What does stuck mean here? Also just to be clear, changing dimensions by any means causes this? Does this occur with just Data Attributes and dependencies installed (or with PlayerEx)?

You're probably (hopefully) right in that this bugged behaviour is likely all a symptom of one dimension-related issue.

**Comment by TheRobberPanda** [comment]
Here is all the information I could gather.

Character gets stuck in **any** dimension after entering it. Killing yourself will not fix the issue. Exiting and entering the world will, however exiting the dimension or entering a new one will make the bug happen again. I happens both in creative and in survival. Here's a [video ](https://youtu.be/eDrGitF9vkE)of what "stuck" means.

- Bug _will not_ happen with levelz installed
- Bug _will not_ happen playerex installed
- Bug _will_ happen with levelz _and_ playerex installed. Here is the [log](https://paste.ee/p/idyXH) stopped right after entering the nether (with the bug)

I can reproduce the bug as many times as needed.

**Comment by CleverNucleus** [comment]
Thanks for the information, this is very helpful. I'll provide updates with any progress.

**Comment by CleverNucleus** [comment]
Hi there,
I've fixed the issue causing the player to become stuck when entering a dimension. 

**About the first bug (Fatal crash upon dying in the nether):** I was unable to reproduce this behaviour; however, the crash log that you provided shows that the error causing the crash was actually the same as the error causing the 'stuck' behaviour that you reported later on. That is to say that, while I was unable to reproduce the initial bug report (I could reproduce the stuck behaviour), I am confident that both behaviours were due to the same core issue - which should now be resolved.

Thanks again, your detailed information was very helpful in expediting the solution.
